pronotepy/dataClasses.py

```diff
@@ -449,6 +449,18 @@ def overall_average(self) -> float:
         else:
             average = -1
         return average
+    
+    @property
+    def group_average(self) -> float:
```

Comment:
Since you're returning `None` if there is no group average, this must be:
```suggestion
    def group_average(self) -> Optional[float]:
```